src/par/FixColumnPartitions.c

```diff
@@ -99,18 +114,71 @@ int FixColumnPartitions_UpDownFaces(Mesh_ptr mesh, MRegion_ptr mr, MFace_ptr* up
     MSTK_Report("FixColumnPartitions","Mesh is not quite columnar, up and down faces do not align.",MSTK_FATAL);
   }
 
-  if (upcen[2] < dncen[2] - 1.e-12) {
-    rf = *up;
-    *up = *dn;
-    *dn = rf;
-    ret = 1;
-  } else if (upcen[2] > dncen[2] + 1.e-12) {
-    ret = 1;
-  } else {
-    /* find the first region upward that is proper */
-    MSTK_Report("FixColumnPartitions","Partitioning with degenerate cells is not yet supported.",MSTK_FATAL);
-    ret = 0;
+  if (up_unknown == 1) {
```

Comment:
Looks like you are recursively traveling up the column and down the column to find a cell that is not degenerate. Wouldn't it be simpler to just check normals? If the z component of the face w.r.t. the cell is greater than 0 it is up, otherwise it is down? But this is fine too